AAInstrumentation/core.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+from subprocess import Popen, PIPE, run, DEVNULL
+import os
+import argparse
+from pathlib import Path, PosixPath
+from dataclasses import dataclass
+
+"""
+    This file contains the core functionality of the AAInstrumentation tool.
+    Specific kinds of instrumentation will inherit from AAInstrumentationDriver
+    and implement their specific functionality.
+"""
+
+
+@dataclass
+class AAInstrumentationDriver:
+
+    """
+    This class is the base class for all instrumentation drivers. It
+    contains the common functionality, like generating the baseline,
+    evaluating results, run a single instrumented compilation, etc. In
+    particular, this contains class variables for all the paths that have to
+    be specified to run the instrumentation on different machines and
+    benchmarks.
+    """
+
+    """
+    The path to the instrumented opt, clang, and llc binaries.
```

Comment:
Regarding python documentation, it's more idiomatic (and better supported by tools) to add the attribute documentation in the class docstring, e.g.: https://github.com/DeadCodeProductions/diopter/blob/main/diopter/compiler.py#L156